salt/_runners/metalk8s_checks.py

```diff
@@ -0,0 +1,246 @@
+    )
+
+    # When upgrading the saltenv should be the "metalk8s-<destination version>"
+    if saltenv != f"metalk8s-{dest_version}":
+        errors.append(
+            f"Invalid saltenv '{saltenv}' consider using 'metalk8s-{dest_version}'"
+        )
+
+    # Check that all nodes are in a supported upgrade path
+    # We only support upgrade from one minor version
+    dest = dest_version.split(".")
+    min_version = f"{dest[0]}.{int(dest[1]) - 1}.0"
+
+    for node_name, node_info in metalk8s_pillar["nodes"].items():
+        if (
+            __salt__["salt.cmd"]("pkg.version_cmp", min_version, node_info["version"])
+            == 1
+        ):
+            errors.append(
+                "Unable to upgrade from more than 1 minor version, Node "
+                f"{node_name} is in {node_info['version']} and you try to upgrade "
+                f"to {dest_version}"
+            )
+
+    nodes_ready = nodes(node_list=metalk8s_pillar["nodes"].keys(), raises=False)
+    if nodes_ready is not True:
+        errors.extend(nodes_ready["errors"])
+
+    minions_ready = minions(minion_list=metalk8s_pillar["nodes"].keys(), raises=False)
+    if minions_ready is not True:
+        errors.extend(minions_ready["errors"])
+
+    return _handle_errors(errors, raises)
+
+
+def downgrade(dest_version, saltenv, raises=True, bypass_disable=False):
+    """Check that we can start MetalK8s cluster downgrade
+
+    Args:
+        dest_version (string): Destination version for downgrade
+        saltenv (string): Salt environment that will be used for downgrade
+        raises (bool, optional): Whether or not this function should raise. Defaults to True.
+        bypass_disable (bool, optional): Whether or not bypass downgrade disable. Default to False.
+
+    Raises:
+        CheckError: If 'raises' is True and some upgrade condition are not met
+
+    Returns:
+        dict or True: An error message or True if everything OK
+    """
+    errors = []
+
+    # NOTE: We use `get_from_map` as we want also defaults from `map.jinja`
+    # to be merged
+    metalk8s_pillar = __salt__["salt.cmd"](
+        "metalk8s.get_from_map", "metalk8s", saltenv=saltenv, with_pillar=True
+    )
+
+    # Check that all nodes are in a supported downgrade path
+    # We only support downgrade from one minor version
+    dest = dest_version.split(".")
+    max_version = f"{dest[0]}.{int(dest[1]) + 2}.0"
+
+    newest_node_version = None
+
+    for node_name, node_info in metalk8s_pillar["nodes"].items():
+        if (
+            newest_node_version is None
+            or __salt__["salt.cmd"](
+                "pkg.version_cmp", node_info["version"], newest_node_version
+            )
+            == 1
+        ):
+            newest_node_version = node_info["version"]
+
+        if (
+            __salt__["salt.cmd"]("pkg.version_cmp", max_version, node_info["version"])
+            != 1
+        ):
```

Comment:
For example, with:
```
dest_version = 2.10.0
node_info["version"] = 2.9.0
```
We end up with `max_version` equal to `2.12.0`, meaning this test will return 1.
```
# salt-call pkg.version_cmp 2.12.0 2.9.0
local:
    1
```
It means we can downgrade to a greater version than the one actually deployed.

Reply:
No it's expected, in this case we will not fail here and in downgrade orchestrate we will just skip this node 
```yaml
Skip node {{ node }}, already in {{ node_version }} older than {{ dest_version }}:
  test.succeed_without_changes
```